src/item_factory.cpp

```diff
@@ -1744,11 +1752,90 @@ void Item_factory::load_pet_armor( const JsonObject &jo, const std::string &src
 
 void islot_armor::load( const JsonObject &jo )
 {
-    optional( jo, was_loaded, "encumbrance", encumber, 0 );
-    // Default max_encumbrance will be set to a reasonable value in
-    // finalize_post
-    optional( jo, was_loaded, "max_encumbrance", max_encumber, -1 );
-    optional( jo, was_loaded, "coverage", coverage, 0 );
+    if( jo.has_array( "coverage_data" ) ) {
+        bool dont_add_first = false;
+        if( !data.empty() ) { // Uses copy-from
+            dont_add_first = true;
+            const JsonObject &obj = *jo.get_array( "coverage_data" ).begin();
+            random_armor_data tempData;
+
+            if( obj.has_array( "encumbrance" ) ) {
+                tempData.encumber = obj.get_array( "encumbrance" ).get_int( 0 );
+                tempData.max_encumber = obj.get_array( "encumbrance" ).get_int( 1 );
+            } else if( obj.has_int( "encumbrance" ) ) {
+                tempData.encumber = obj.get_int( "encumbrance" );
+                tempData.max_encumber = 0;
+            }
+            tempData.coverage = obj.get_int( "coverage" );
+            data[0].encumber = tempData.encumber;
+            data[0].max_encumber = tempData.max_encumber;
+            data[0].coverage = tempData.coverage;
+        }
+
+        for( const JsonObject &obj : jo.get_array( "coverage_data" ) ) {
+            // If this item used copy-from, data[0] is already set, so skip adding first data
+            if( dont_add_first ) {
+                dont_add_first = false;
+                continue;
+            }
+            random_armor_data tempData;
+            body_part_set temp_cover_data;
+            assign_coverage_from_json( obj, "parts", temp_cover_data, sided );
+            tempData.covers = temp_cover_data;
+
+            if( obj.has_array( "encumbrance" ) ) {
+                tempData.encumber = obj.get_array( "encumbrance" ).get_int( 0 );
+                tempData.max_encumber = obj.get_array( "encumbrance" ).get_int( 1 );
+            } else if( obj.has_int( "encumbrance" ) ) {
+                tempData.encumber = obj.get_int( "encumbrance" );
+                tempData.max_encumber = 0;
+            }
+            tempData.coverage = obj.get_int( "coverage" );
+            data.push_back( tempData );
+
+            if( obj.has_string( "layer" ) ) {
+                for( auto &piece : data ) {
+                    const std::string &val = obj.get_string( "layer" );
+                    auto iter = std::find(layer_level_strings.begin(), layer_level_strings.end(), val );
+                    if( iter != layer_level_strings.end() ) {
+                        piece.layer = static_cast<layer_level>( iter - layer_level_strings.begin() );
+                    }
```

Comment:
You're looking up the same `layer_level` level value in every loop iteration; it should be pulled outside.

Moreover, if you implement the `io::` infrastruture for this enum then you can simply do
```
layer_level layer;
obj.read( "layer", layer )
```
and it will do all this under the hood.

For that you'll need to specialise `enum_traits` and implement `io::enum_to_string` for `layer_level`.  You can see how other enums do this.